crates/ark/src/interface.rs

```diff
@@ -620,6 +620,19 @@ impl RMain {
                 log::error!("Error registering some hooks: {err:?}");
             }
 
+            // Set .Platform$GUI to "Positron" for console sessions only
+            if session_mode == SessionMode::Console {
+                let result = harp::parse_eval_global(r#"
+                    unlockBinding(".Platform", baseenv())
+                    .Platform$GUI <- "Positron"
+                    lockBinding(".Platform", baseenv())
+                "#);
+                match result {
+                    Ok(_) => log::info!("Set .Platform$GUI to 'Positron' for console session"),
+                    Err(err) => log::warn!("Failed to set .Platform$GUI: {err:?}"),
+                }
+            }
+
```

Comment:
I'm not sure what the best approach is, but this one is at least easy to reason about.

Some other approaches I tried out included:

- Do something like `r_symbol!(".Platform")` in Rust and find the `GUI` element. I like that this doesn't involve executing R code during some initialization step, but you still have to lock/unlock (just at the C level).
- Do the locking/unlocking in `startup.R`. This feels like a better place to run such R code, but I'm not sure we can check session mode there and the other options there are more user-facing.

RStudio does it here, which is not exactly the same but similar in that in happens post-initialization:
https://github.com/rstudio/rstudio/blob/5381a9a4363eae00007fd170302fa8a5c93050b5/src/cpp/r/R/Options.R#L126